src/t8_cmesh/t8_cmesh_examples.c

```diff
@@ -2527,6 +2547,141 @@ t8_cmesh_new_full_hybrid (sc_MPI_Comm comm)
   return cmesh;
 }
 
+
+t8_cmesh_t
+t8_cmesh_new_pyramid_cake (sc_MPI_Comm comm, int num_of_pyra)
+{
+  
+  int                 i, j;
+  double             *vertices = T8_ALLOC (double, num_of_pyra * 5 * 3);
+  t8_cmesh_t          cmesh;
+  double              degrees = 360. / num_of_pyra;
+  if (vertices)
+    T8_ASSERT (num_of_pyra > 2);
```

Comment:
```suggestion
  T8_ASSERT (num_of_pyra > 2);
```
I do not know why this check shouldn't be executed without checking of the vertices